File: backend/schemas/market_overview_scanner.py
"""Schemas for market overview scanner — companies with biggest metric changes."""

from pydantic import BaseModel


class ScannerRow(BaseModel):
    company_name: str
    industry: str | None = None
    market_cap: float | None = None
    q3_ago: float | None = None
    q2_ago: float | None = None
    q1_ago: float | None = None
    current_qtr: float | None = None
    change_3q: float | None = None


class ScannerResponse(BaseModel):
    metric: str
    metric_label: str
    title: str
    subtitle: str
    periods: list[str]
    period_type: str  # "quarterly" or "annual"
    rows: list[ScannerRow]

File: frontend-next/src/types/marketOverviewScanner.ts
export interface ScannerRow {
  company_name: string;
  industry: string | null;
  market_cap: number | null;
  q3_ago: number | null;
  q2_ago: number | null;
  q1_ago: number | null;
  current_qtr: number | null;
  change_3q: number | null;
}

export interface ScannerResponse {
  metric: string;
  metric_label: string;
  title: string;
  subtitle: string;
  periods: string[];
  period_type: string;
  rows: ScannerRow[];
}
